### PR TITLE
Make global ComWrappers for marshalling respect requested IID

### LINKED ISSUE #35883: ComWrappers do not respect interface IID when used for marshalling
See https://github.com/dotnet/corert/issues/4219#issuecomment-624461282 for details

### PR DESCRIPTION
**Pull request by elinor-fung**
cc @AaronRobinsonMSFT @jkoritzinsky 

Resolves #35883

### CODE REVIEW

File: src/coreclr/src/vm/interopconverter.cpp [hunk 1]
```diff
@@ -466,9 +480,11 @@ IUnknown *GetComIPFromObjectRef(OBJECTREF *poref, REFIID iid, bool throwIfNoComI
     {
         SafeComHolder<IUnknown> pvObj;
         hr = pUnk->QueryInterface(iid, &pvObj);
+        pUnk->Release();
         if (FAILED(hr))
             COMPlusThrowHR(hr);
 
+        pUnk = pvObj.Extract();
```
Comment (AaronRobinsonMSFT) [inline src/coreclr/src/vm/interopconverter.cpp:487]: I would avoid reusing the the `pUnk` in this case. Perhaps just `RETURN (pvObj.Extract());`?

File: src/coreclr/src/vm/interopconverter.cpp [hunk 2]
```diff
@@ -194,6 +202,12 @@ IUnknown *GetComIPFromObjectRef(OBJECTREF *poref, ComIpType ReqIpType, ComIpType
         if (pFetchedIpType != NULL)
             *pFetchedIpType = ReqIpType;
 
+        if (pvObj != NULL)
```
Comment (AaronRobinsonMSFT) [inline src/coreclr/src/vm/interopconverter.cpp:205]: Let's fold this into a new `else` block in the `if`/`else if` chain above. That way the `pvObj` can always represent the return.

File: src/coreclr/src/vm/interopconverter.cpp (hunk 1 above)
Comment (jkotas) [reply src/coreclr/src/vm/interopconverter.cpp:487]: You do not really need `SafeComHolder` here at all.

Comment (jkotas) [reply src/coreclr/src/vm/interopconverter.cpp:487]: (`pUnk` does not use it either.)

File: src/coreclr/src/vm/interopconverter.cpp [hunk 3]
```diff
@@ -466,9 +480,11 @@ IUnknown *GetComIPFromObjectRef(OBJECTREF *poref, REFIID iid, bool throwIfNoComI
     {
         SafeComHolder<IUnknown> pvObj;
         hr = pUnk->QueryInterface(iid, &pvObj);
```
Comment (jkotas) [inline src/coreclr/src/vm/interopconverter.cpp:482]: Do we need to switch to preemptive mode around the `QueryInterface` ?

**Review by jkotas — APPROVED** [review]

**Review by AaronRobinsonMSFT — APPROVED** [review]

**Pull request merged**